### ISSUE TITLE
[Bug] Unsupported table metadata field type 0 when parsing binlog event

### ISSUE THREAD

**Issue by EchoLee5** [ververica/flink-cdc-connectors#2784]
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Flink version

1.14.5

### Flink CDC version

2.4.0

### Database and its version

mysql 8.0.22-13

### Minimal reproduce step

```
CREATE TABLE `table1` (`c1` int(20) NOT NULL AUTO_INCREMENT, `c2` year, `c3` int, `c4` int, `c5` int, `c6` int, `c7` int, `c8` int, `c9` int, c10 char(10), PRIMARY KEY (`c1`));
INSERT INTO `table1` (c1, c2, c3, c4, c5, c6, c7, c8, c9, c10) VALUES (1, 2, 3, 4, 5, 6, 7, 8, 9, '');
```

```
CREATE TABLE flink_cdc(
  `c1` int,
  `c2` int ,
  `c3` int ,
  `c4` int ,
  `c5` int ,
  `c6` int ,
  `c7` int ,
  `c8` int ,
  `c9` int ,
  `c10` char(10) ,
  PRIMARY KEY (`c1`)  NOT ENFORCED
) WITH (
    'connector' = 'mysql-cdc',
       'hostname' = 'xxx',
       'port' = '33061',
       'username' = 'xxx',
       'password' = 'xxx',
       'database-name' = 'test',
       'table-name' = 'table1',
       'scan.startup.mode' = 'specific-offset',
       'scan.startup.specific-offset.file' = 'mysql-bin.000353',
       'scan.startup.specific-offset.pos' = '4'
);

CREATE TABLE print (
  `c1` int,
  `c2` int ,
  `c3` int ,
  `c4` int ,
  `c5` int ,
  `c6` int ,
  `c7` int ,
  `c8` int ,
  `c9` int ,
  `c10` char(10) ,
  PRIMARY KEY (`c1`)  NOT ENFORCED
) WITH (
   'connector' = 'print'
);

insert into print select * from flink_cdc;
```

### What did you expect to see?

Data can be synchronized normally

### What did you see instead?

```
Caused by: java.io.IOException: Unsupported table metadata field type 0
	at com.github.shyiko.mysql.binlog.event.deserialization.TableMapEventMetadataDeserializer.deserialize(TableMapEventMetadataDeserializer.java:52) ~[mysql-binlog-connector-java-0.27.2.jar:0.27.2]
	at com.github.shyiko.mysql.binlog.event.deserialization.TableMapEventDataDeserializer.deserialize(TableMapEventDataDeserializer.java:59) ~[flink-connector-mysql-cdc-2.4.0.jar:0.27.2]
	at com.github.shyiko.mysql.binlog.event.deserialization.TableMapEventDataDeserializer.deserialize(TableMapEventDataDeserializer.java:37) ~[flink-connector-mysql-cdc-2.4.0.jar:0.27.2]
	at com.github.shyiko.mysql.binlog.event.deserialization.EventDeserializer.deserializeEventData(EventDeserializer.java:335) ~[mysql-binlog-connector-java-0.27.2.jar:0.27.2]
	at com.github.shyiko.mysql.binlog.event.deserialization.EventDeserializer.deserializeTableMapEventData(EventDeserializer.java:313) ~[mysql-binlog-connector-java-0.27.2.jar:0.27.2]
	at com.github.shyiko.mysql.binlog.event.deserialization.EventDeserializer.nextEvent(EventDeserializer.java:237) ~[mysql-binlog-connector-java-0.27.2.jar:0.27.2]
	at io.debezium.connector.mysql.MySqlStreamingChangeEventSource$1.nextEvent(MySqlStreamingChangeEventSource.java:260) ~[flink-connector-mysql-cdc-2.4.0.jar:1.9.7.Final]
	at com.github.shyiko.mysql.binlog.BinaryLogClient.listenForEventPackets(BinaryLogClient.java:1051) ~[mysql-binlog-connector-java-0.27.2.jar:0.27.2]
	at com.github.shyiko.mysql.binlog.BinaryLogClient.connect(BinaryLogClient.java:631) ~[mysql-binlog-connector-java-0.27.2.jar:0.27.2]
	at com.github.shyiko.mysql.binlog.BinaryLogClient$7.run(BinaryLogClient.java:932) ~[mysql-binlog-connector-java-0.27.2.jar:0.27.2]
```

### Anything else?
[https://github.com/osheroff/mysql-binlog-connector-java/issues/108](url)

![1701323018411-f5d0cc88-288d-4b7f-ac3b-c056e3bcadbc](https://github.com/ververica/flink-cdc-connectors/assets/39044001/0ccf6adc-273a-4cf4-85cb-419665d9a2d9)
![1701323111118-0cff40f3-13d0-4471-8932-071cdbf46d8f](https://github.com/ververica/flink-cdc-connectors/assets/39044001/8f8f26a1-a6ad-4a34-ba95-74fd10b21510)

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by EchoLee5** [comment]
@leonardBang Can you help me look into this issue?

**Comment by ruanhang1993** [comment]
Fixed in #2682 together.
